### PR TITLE
Improved "quick launch arguments" GUI and internal handling

### PR DESCRIPTION
**Pull request by RevZero**
Replaced check-boxes for quick launch arguments with drop-down-list. 
Added additional arguments (other suggestions?). We could add tool tips to the drop down list, if someone provides a one-liner for each argument, explaining its function.

2019-06-09 00_35_05-Window

See commit message and code comments for details.

### CODE REVIEW

File: xcom2-launcher/xcom2-launcher/Classes/Settings.cs [hunk 1]
```diff
@@ -53,9 +56,20 @@ public string GamePath
...
+        /// List of XCOM Comamnd line arguments that will be used when the game is started.
+        /// </summary>
+        public ReadOnlyCollection<string> ArgumentList
+        {
+            get => _argumentList.AsReadOnly();
+            set => _argumentList = value.ToList();
+        }
 
-        public bool CheckForUpdates { get; set; } = true;
+		public bool CheckForUpdates { get; set; } = true;
```
Comment (Xymanek) [inline xcom2-launcher/xcom2-launcher/Classes/Settings.cs]: Again whitespace issues

File: xcom2-launcher/xcom2-launcher/Forms/MainForm.Events.cs [hunk 2]
```diff
@@ -141,6 +141,17 @@ internal void RegisterEvents()
...
 
+        private void QuickArgumentItemClick(object sender, EventArgs e) {
+            // Add or remove the respective argument from the argument list, depending on its check-state.
+            if (sender is ToolStripMenuItem item)
+            {
+                if (item.Checked)
+                    Settings.AddArgument(item.Text);
+                else
+                    Settings.RemoveArgument(item.Text);
+            }
```
Comment (Xymanek) [inline xcom2-launcher/xcom2-launcher/Forms/MainForm.Events.cs]: Maybe it makes sense to `else { throw }` or something? Dunno how it's usually done in winforms

File: xcom2-launcher/xcom2-launcher/Classes/Settings.cs (hunk 1 above)
Comment (RevZero) [inline xcom2-launcher/xcom2-launcher/Classes/Settings.cs]:
```suggestion
        public bool CheckForUpdates { get; set; } = true;
```

File: xcom2-launcher/xcom2-launcher/Forms/MainForm.Events.cs (hunk 2 above)
Comment (RevZero) [reply xcom2-launcher/xcom2-launcher/Forms/MainForm.Events.cs]: Probably a matter of taste in this case. But can not hurt, so someone who uses the handler in a wrong way gets "notified". Would actually be a good use case for an "Assertion". I will change it.

**Pull request merged**